tools/CreateNuget.ps1

```diff
@@ -68,17 +94,179 @@ function UpdateDependencyVersions([string]$Nuspec)
+    # for different frameworks (i.e. ReactiveUI). Therefore if our search
+    # returns more than one node, then we take the one that matches 
+    # the Framework in its Condition
+
+    if ($targetPackage.Node.Count -gt 1)
+    {
+        foreach ($tn in $targetPackage.Node)
+        {
+            if ($tn.Condition -match $Framework )
+            {
+                $currentCondition = $tn.Condition
+                $currentVersion = $tn.Version
+            }
+        }
+    }
+    else
+    {
+        $currentCondition = $targetPackage.Node.Condition
+        $currentVersion = $targetPackage.Node.Version
+    }
+
+    if ($currentCondition -match "==")
+    {
+        $compOperator = "=="
+    }
+
+    if ($currentCondition -match "!=")
+    {
+        $compOperator = "!="
+    }
+
+    if ($currentCondition -match "net452")
+    {
+        $currentFramework = "net452"
+    }
+
+    if ($currentCondition -match "net472")
+    {
+        $currentFramework = "net472"
+    }
+
+    if ($currentCondition -match "netstandard2.0")
+    {
+        $currentFramework = "netstandard2.0"
+    }
+
+    $myObj = New-Object -TypeName PackagRef 
+    $myObj.Version = $currentVersion
+    $myObj.ComparisonOperator = $compOperator 
+    $myObj.Framework = $currentFramework
+    
+    return $myObj
+}
+
+# UpdateDependencyVersions
+#
+#    Helper function that updates all non-ReactiveDomain dependencies 
+#    in a nuspec file. Loops through all dependencies listed in a 
+#    nuspec file and gets the versions from its
+#    entry in the corresponding .csproj file
+#
+function UpdateDependencyVersions([string]$Nuspec, [string]$CsProj)
+{
+    Write-Host "Updating dependency versions of: " $Nuspec
+
+    [xml]$xml = Get-Content -Path $Nuspec -Encoding UTF8
+    $dependencyNodes = $xml.package.metadata.dependencies.group.dependency
+
+
+    $f452 = $xml | Select-XML -XPath "//package/metadata/dependencies/group[@targetFramework='.NETFramework4.5.2']"
+    $framework452Nodes = $f452.Node.ChildNodes
+
+    $f472 = $xml | Select-XML -XPath "//package/metadata/dependencies/group[@targetFramework='.NETFramework4.7.2']"
+    $framework472Nodes = $f472.Node.ChildNodes
+
+    $netstandard2 = $xml | Select-XML -XPath "//package/metadata/dependencies/group[@targetFramework='.NETStandard2.0']"
+    $netstandard2Nodes = $netstandard2.Node.ChildNodes
+
+    foreach($refnode in $framework452Nodes)
+    {
+        if ( $refnode.id -match "ReactiveDomain"){continue}
+
+        $pRef = GetPackageRefFromProject $refnode.id $CsProj "net452"
+        if ((($pRef.ComparisonOperator -eq "" -or $pRef.Framework -eq "") -or 
+            ($pRef.ComparisonOperator -eq "==" -and $pRef.Framework -eq "net452") -or 
+            ($pRef.ComparisonOperator -eq "!=" -and $pRef.Framework -ne "net452")) -and
+            ($pRef.version -ne ""))
+        {
+            $refnode.version = $pRef.Version
+        }       
+    }
+
+    foreach($refnode in $framework472Nodes)

```

Comment:
We'll have the same problem here with updated versions of .NET. Can we somehow collapse these, maybe with a list of framework nodes that we iterate over?

Reply:
see above